### ISSUE TITLE
Max line length in read_data

### ISSUE THREAD

**Issue by bendjensen** [lammps/lammps#2735]
There appears to be a maximum line length permitted in a read_data file of 255 characters. The most recent master branch LAMMPS (8 Apr 2021) will produce the following misleading error message if the line length exceeds this limit: "ERROR: Unexpected empty line in BondCoeffs section (src/read_data.cpp:1829)" The most recent unstable branch LAMMPS (8 Apr 2021) produces a segmentation fault.

The error can be reproduced by adding characters to the dreiding example:
```
diff  examples/dreiding/data.dreiding examples/dreiding/data.dreiding_error
29c29
<     1             350            1.42 # O_3 C_3
---
>     1             350            1.42 # O_3 C_3 aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaeaaaaaaaaaaaaaaaaa
```

I dug around the manual and didn't see a reference to a maximum line length in a read_data file, perhaps it could be added to the manual. It could also be helpful to users if the error message indicated the cause was a line that was too long.

I'm currently converting a data file to LAMMPS format and trying to retain as much of the parameter provenance as  comments in the file, which is causing the lines to exceed the limit in some cases.

**Comment by akohlmey** [comment]
@bendjensen yes, there is a compile time limit of 256 characters (that is 255 characters plus the terminating NULL).

https://github.com/lammps/lammps/blob/3455172f7d50f6871202d01d000a94f2283ead35/src/read_data.cpp#L44

You can change this define to use a larger value and recompile and it should be fine.

This part of LAMMPS is about to be refactored which would increase the maximum line length to 1023 characters (due to an internal limit of the new facility).

The detection of longer lines would require extra consideration, but is non-trivial because of the parallel nature of LAMMPS and the fact that the data file can be very large with many atoms and thus has to be read, communicated and processed in relatively small chunks. What can be done more easily is to *truncate* such extra long lines (i.e. read them completely but do not communicate and process anything beyond the initial part.

**Comment by bendjensen** [comment]
@akohlmey Thank you for the clarification, and point me to the define in read_data.cpp. If LAMMPS can truncate the line that would work just fine in this use-case since comments are producing the excessively long lines and can safely be ignored.

**Comment by akohlmey** [comment]
@bendjensen truncation is now implemented in pull request #2738 and will be available in the next LAMMPS patch release.

**Comment by bendjensen** [comment]
@akohlmey Thank you!
